org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -284,6 +297,53 @@ public void didChangeConfiguration(DidChangeConfigurationParams params) {
 		logInfo(">>New configuration: " + settings);
 	}
 
+	private boolean configureVM() throws CoreException {
+		String javaHome = preferenceManager.getPreferences().getJavaHome();
+		if (javaHome != null) {
+			File jvmHome = new File(javaHome);
+			if (jvmHome.isDirectory()) {
+				IVMInstall defaultVM = JavaRuntime.getDefaultVMInstall();
+				File location = defaultVM.getInstallLocation();
+				if (!location.equals(jvmHome)) {
+					IVMInstall vm = findVM(jvmHome);
+					if (vm == null) {
+						IVMInstallType installType = JavaRuntime.getVMInstallType(StandardVMType.ID_STANDARD_VM_TYPE);
+						long unique = System.currentTimeMillis();
+						while (installType.findVMInstall(String.valueOf(unique)) != null) {
+							unique++;
+						}
+						String vmId = String.valueOf(unique);
+						VMStandin vmStandin = new VMStandin(installType, vmId);
+						String name = jvmHome.getName();
+						name = name.trim();
+						if (name.length() == 0) {
+							name = "JRE";
+						}
+						vmStandin.setName(name);
+						vmStandin.setInstallLocation(jvmHome);
+						vm = vmStandin.convertToRealVM();
+					}
+					JavaRuntime.setDefaultVMInstall(vm, new NullProgressMonitor());
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	private IVMInstall findVM(File jvmHome) {
+		IVMInstallType[] types = JavaRuntime.getVMInstallTypes();
+		for (IVMInstallType type : types) {
+			IVMInstall[] installs = type.getVMInstalls();
+			for (IVMInstall install : installs) {
+				if (jvmHome.equals(install.getInstallLocation())) {
```

Comment:
are we sure the jvmHome passed as a parameter will have the same format as install.getInstallLocation()? i.e. no funny business with back/forward slashes?

Reply:
I am comparing files (java.io.File), not strings or paths so any format isn't important.

Reply:
ok fair enough, I would have noticed if I reviewed the code from the IDE.